src/test/java/org/folio/rest/TestUtil.java

```diff
@@ -86,9 +87,9 @@ public JsonObject getJson() {
           System.out.println(String.format(
                   "Got new WrappedResponse object with values code %s, body %s",
                   wr.getCode(), wr.getBody()));
-          if(!future.isComplete() && !future.failed()) {
+          if(!promise.future().isComplete() && !promise.future().failed()) {
```

Comment:
Could `promise.future()` be extracted to a variable. Is it guaranteed that the future will be the same each time this is invoked?

Reply:
Extracted to variable.

>  Is it guaranteed that the future will be the same each time this is invoked?

I think it is.